lib/src/model_utils.dart

```diff
@@ -108,20 +109,16 @@ bool hasPrivateName(Element e) {
       return true;
     }
   }
-  if (e is LibraryElement &&
-      (e.identifier.startsWith('dart:_') ||
-          e.identifier.startsWith('dart:nativewrappers/') ||
-          ['dart:nativewrappers'].contains(e.identifier))) {
-    return true;
-  }
   if (e is LibraryElement) {
-    var elementLocation = e.location;
-    if (elementLocation != null) {
-      var locationParts = elementLocation.components[0].split(slashes);
-      // TODO(jcollins-g): Implement real cross package detection
-      if (locationParts.length >= 2 &&
-          locationParts[0].startsWith('package:') &&
-          locationParts[1] == 'src') return true;
+    if (e.identifier.startsWith('dart:_') ||
+        e.identifier.startsWith('dart:nativewrappers/') ||
+        'dart:nativewrappers' == e.identifier) {
+      return true;
+    }
+    var elementUri = e.source.uri;
+    // TODO(jcollins-g): Implement real cross package detection
+    if (elementUri.scheme == 'package' && elementUri.pathSegments[1] == 'src') {
```

Comment:
Practically, any `UriResolver` that understands `package:` URIs will return `null` if there are less than `2` segments, so this code should be fine. But purely from `Uri` point of view, it is possible to parse `package:foo`. So, if we could make it safer... Something like `elementAtOrNull`, similar to `firstOrNull` from `IterableExtension` from `package:collection`. Unfortunately it does not exist yet :-)

Anyway, just some random notes.

Reply:
Yeah I decided the simplicity was worth ignoring `package:foo`, haha. We'll see if it ever causes a problem.